logging/logging.go

```diff
@@ -894,7 +899,23 @@ func deconstructXCloudTraceContext(s string) (traceID, spanID string, traceSampl
 	return
 }
 
-func (l *Logger) toLogEntry(e Entry) (*logpb.LogEntry, error) {
+// ToLogEntry takes an Entry structure and converts it to the LogEntry proto.
+// A parent can take any of the following forms:
+//    projects/PROJECT_ID
+//    folders/FOLDER_ID
+//    billingAccounts/ACCOUNT_ID
+//    organizations/ORG_ID
+// for backwards compatibility, a string with no '/' is also allowed and is interpreted
+// as a project ID.
+//
+// ToLogEntry is implied when users invoke Logger.Log or Logger.LogSync,
+// but its exported as a pub function here to give users additional flexibility
+// when using the library
+func ToLogEntry(e Entry, parent string) (*logpb.LogEntry, error) {
+	return toLogEntryInternal(e, nil, makeParent(parent))
+}
+
+func toLogEntryInternal(e Entry, client *Client, parent string) (*logpb.LogEntry, error) {
```

Comment:
If we think there is a big enough use case for something like this I wonder if we should make an optional Parent field on `Entry`? It could fall back to the parent specified on the client if not set.

Reply:
Afraid that it can be a little bit dangerous. Unless you think that's completely OK to fail ToLogEntry if we need this field for traceID?

Reply:
Entry already has `parent` as a substring in its `LogName` field.

Might I suggest creating `ToLogEntry` as a method on `Entry`: 
`func (e *Entry) ToLogEntry(client ...*Client)`. It's still not dependent on client like you wanted. And function signature is cleaner, and we dont need to create new internal functions.

Reply:
LogName can't be used, because toLogEntry is failing whenever we set that field. Unless you want me to remove that check?

Reply:
I think it makes sense to remove that check in cases where there's no client/logger. And to update the corresponding comments. Ofc when there is a logger/client, then users can't set the logname (same behavior as before). 

Also LogName and Resource fields are required in order to write to [Cloud Logging](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry). As a part of your implementation, can you also write checks to ensure that if ppl use ToLogEntry, they are creating valid LogEntries?

Reply:
Strange, I have received a comment, but don't see it here.
> I think it makes sense to remove that check in cases where there's no client/logger. And to update the corresponding comments. Ofc when there is a logger/client, then users can't set the logname (same behavior as before).
Also LogName and Resource fields are required in order to write to Cloud Logging. As a part of your implementation, can you also write checks to ensure that if ppl use ToLogEntry, they are creating valid LogEntries?

1. (Related to previous comment). Golang doesn't support optional arguments. `client ...*Client` means that  we can pass multiple clients and that's a very confusing in this context.
2. I think different behavior when you call with Client and without is also very confusing.
3. LogName and resource are optional. They can be set on WriteLogEntries level or even ignored completely.

Reply:
I deleted this comment because after thinking through various modifications to make this PR work, I don't think this logic should be introduced at the client library at this veneer layer. Agree that using a variadic param to hack things isn't the right way to go.